tests/test_screenshot.py

```diff
@@ -99,3 +99,58 @@ async def test_screenshot(websocket, context_id, png_base64):
     img1 = Image.open(io.BytesIO(base64.b64decode(resp["result"]["data"])))
     img2 = Image.open(io.BytesIO(base64.b64decode(png_base64)))
     assert_images_equal(img1, img2)
+
+
+@pytest.mark.asyncio
+async def test_screenshot_oopif(websocket, context_id, html, iframe):
+    await goto_url(websocket,
+                   context_id,
+                   html(iframe("https://www.example.com")),
+                   wait="complete")
+
+    iframe_context_id = (await get_tree(
+        websocket, context_id))["contexts"][0]["children"][0]["context"]
+    assert iframe_context_id != context_id
+
+    command_result = await execute_command(websocket, {
+        "method": "cdp.getSession",
+        "params": {
+            "context": context_id
+        }
+    })
+    session_id = command_result["cdpSession"]
+
+    # Set a fixed viewport to make the test deterministic.
+    await execute_command(
+        websocket, {
+            "method": "cdp.sendCommand",
+            "params": {
+                "cdpMethod": "Emulation.setDeviceMetricsOverride",
+                "cdpParams": {
+                    "width": 200,
+                    "height": 200,
+                    "deviceScaleFactor": 1.0,
+                    "mobile": False,
+                },
+                "cdpSession": session_id
+            }
+        })
+
+    await send_JSON_command(
+        websocket, {
+            "method": "browsingContext.captureScreenshot",
+            "params": {
+                "context": iframe_context_id
+            }
+        })
+
+    resp = await read_JSON_message(websocket)
+    assert resp["result"] == {'data': ANY_STR}
+
+    img1 = Image.open(io.BytesIO(base64.b64decode(resp["result"]["data"])))
+    img2 = Image.open(
+        io.BytesIO(
+            base64.b64decode(
```

Comment:
This assertion can be platform-dependent, as e.g. fonts can be rendered in different ways.

Reply:
This is not different from the status quo: https://github.com/GoogleChromeLabs/chromium-bidi/blob/1ed0b93ec87854752221f44a70327cc4089b7b4f/tests/test_screenshot.py#L53

I am open to platform-independent ideas if you have any suggestion though.